openedx/openedx-platform
# feat!: Remove outdated Libraries Relaunch cruft

## Deadline

ASAP, since:

* We don't want LibraryContentBlock to quietly (and buggily) support sync-from-V2-libs in the MVP
* This greatly simplifies: https://github.com/openedx/edx-platform/pull/35553. The ItemBankBlock, which we want in the MVP, will support randomized problem banks with V2 library content. The ItemBankBlock will be the base class of the LegacyLibraryContentBlock, simplifying our future content migration.

## Description & supporting info

The V2 libraries project had a few past iterations which were never launched. This commit cleans up pieces from those which we don't need for the real Libraries Relaunch MVP in Sumac:

* Remove ENABLE_LIBRARY_AUTHORING_MICROFRONTEND, LIBRARY_AUTHORING_FRONTEND_URL, and REDIRECT_TO_LIBRARY_AUTHORING_MICROFRONTEND, all of which are obsolete now that library authoring has been merged into https://github.com/openedx/frontend-app-authoring. More details on the new Content Libraries configuration settings are here: https://github.com/openedx/frontend-app-authoring/issues/1334

* Remove dangling support for syncing V2 (learning core-backed) library content using the LibraryContentBlock. This code was all based on an older understanding of V2 Content Libraries, where the libraries were smaller and versioned as a whole rather then versioned by-item. Reference to V2 libraries will be done on a per-block basis using the upstream/downstream system, described here: https://github.com/openedx/edx-platform/blob/master/docs/decisions/0020-upstream-downstream.rst It's important that we remove this support now so that OLX course authors don't stuble upon it and use it, which would be buggy and complicate future migrations.

* Remove the "mode" parameter from LibraryContentBlock. The only supported mode was and is "random". We will not be adding any further modes. Going forward for V2, we will have an ItemBank block for randomizing items (regardless of source), which can be synthesized with upstream referenced as described above. Existing LibraryContentBlocks will be migrated.

* Finally, some renamings:

  * LibraryContentBlock -> LegacyLibraryContentBlock
  * LibraryToolsService -> LegacyLibraryToolsService
  * LibrarySummary -> LegacyLibrarySummary

  Module names and the old OLX tag (library_content) are unchanged.

Closes: https://github.com/openedx/frontend-app-authoring/issues/1115

## Testing instructions

* Create a Legacy Library in Studio
  * Add some problems to it
* Create a V2 library in Studio
  * Publish at least one component from it
* Add Library Content to a course unit
  * In the drop-down for selecting a lib, ensure that your legacy library is visible, but your V2 library is not.
  * Select your legacy library
  * Go to "View All", ensure all your problems are there, and make some title and content edits
* Publish your course, view the content in the LMS, and interact with the problems to ensure they work.
* Make an edit to your legacy library
* In your course, ensure that you can "Update Now", and do so
  *  Hit "View All" again, ensuring that your title changes are preserved, but your content edits are erased, and the new content has been pulled in.
* Publish your course, and view the content in LMS again.

## Other information

N/A
